File: common/src/main/java/org/valkyrienskies/clockwork/ClockWorkContraptions.java
package org.valkyrienskies.clockwork;

import com.simibubi.create.content.contraptions.components.structureMovement.ContraptionType;
import org.valkyrienskies.clockwork.content.contraptions.flap.contraption.FlapContraption;
import org.valkyrienskies.clockwork.content.contraptions.propellor.PropellorContraption;

public class ClockWorkContraptions {
    public static final ContraptionType
            FLAP = ContraptionType.register(ClockWorkMod.asResource("flap").toString(), FlapContraption::new),
            PROPELLOR = ContraptionType.register(ClockWorkMod.asResource("propellor").toString(), PropellorContraption::new);

    public static void init() {
    }

}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/ClockworkContraptions.kt
package org.valkyrienskies.clockwork

import com.simibubi.create.content.contraptions.ContraptionType
import org.valkyrienskies.clockwork.content.contraptions.flap.contraption.FlapContraption
import org.valkyrienskies.clockwork.content.contraptions.propeller.contraption.PropellerContraption

object ClockworkContraptions {
    val FLAP = ContraptionType.register(
        ClockworkMod.asResource("flap").toString()
    ) { FlapContraption() }
    val PROPELLER = ContraptionType.register(
        ClockworkMod.asResource("propeller").toString()
    ) { PropellerContraption() }

    fun init() {}
}
